tests/unit/test_operator.py

```diff
@@ -50,6 +65,7 @@ def test_with_relation(harness):
             "password": "",
         },
     )
+    harness.update_config({"public-url": "10.64.140.43.nip.io"})
```

Comment:
I'm debating this because we deploy the charm w/o any configuration for public-url, but is there a way we can det this by default in the harness fixture so we don't have to update this config on every test case? Then in the test case that verifies the unit goes to blocked we can invert the logic.

Reply:
Good idea, pushed a commit for this.